metadata/metadata.go

```diff
@@ -832,6 +866,45 @@ func (resource *modelResource) UpdateSchedule(schedule string) error {
 	return fmt.Errorf("not implemented")
 }
 
+func (resource *modelResource) Update(lookup ResourceLookup) error {
+	val, err := lookup.Lookup(resource.ID())
+	if err != nil {
+		return err
+	}
+	deserialized := val.Proto()
+	model, ok := deserialized.(*pb.Model)
+	if !ok {
+		return errors.New("failed to deserialize existing model record")
+	}
+	updated, err := unionNameVariants(resource.serialized.Features, model.Features)
+	if err != nil {
+		return err
+	}
+	resource.serialized.Features = updated
+	updated, err = unionNameVariants(resource.serialized.Trainingsets, model.Trainingsets)
+	if err != nil {
+		return err
+	}
+	resource.serialized.Trainingsets = updated
+	return nil
+}
+
+func unionNameVariants(source, destination []*pb.NameVariant) ([]*pb.NameVariant, error) {
```

Comment:
Should make a test for this function that tests duplicates, different name variants, nil values, etc